File: backend/pm-requirement/src/main/java/com/pm/requirement/domain/Requirement.java
package com.pm.requirement.domain;

import com.baomidou.mybatisplus.annotation.TableName;
import com.pm.common.base.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

/**
 * 需求实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("requirement")
@Schema(description = "需求")
public class Requirement extends BaseEntity {

    @Schema(description = "所属项目ID")
    private Long projectId;

    @Schema(description = "需求标题")
    private String title;

    @Schema(description = "需求描述")
    private String description;

    @Schema(description = "验收标准")
    private String acceptanceCriteria;

    @Schema(description = "优先级：critical/high/medium/low")
    private String priority;

    @Schema(description = "状态：draft/reviewing/approved/scheduled/done/rejected")
    private String status;

    @Schema(description = "来源：business/tech/regulatory")
    private String source;

    @Schema(description = "分类：general/differential（通用/差异化）")
    private String category;

    @Schema(description = "负责人ID")
    private Long assignedTo;

    @Schema(description = "关联里程碑ID")
    private Long milestoneId;

    @Schema(description = "预估工时（小时）")
    private BigDecimal estimatedHours;

    @Schema(description = "实际工时（小时）")
    private BigDecimal actualHours;

    @Schema(description = "备注")
    private String remark;
}

File: frontend/src/types/requirement.ts
/** 需求状态 */
export type RequirementStatus =
  | 'draft'
  | 'reviewing'
  | 'approved'
  | 'rejected'
  | 'scheduled'
  | 'done'

/** 需求优先级 */
export type Priority = 'critical' | 'high' | 'medium' | 'low'

/** 需求信息（对齐后端 RequirementVO） */
export interface Requirement {
  id: number
  projectId: number
  title: string
  description?: string
  priority: Priority
  status: RequirementStatus
  source?: string
  category?: string
  assignedTo?: number
  milestoneId?: number
  createdBy: number
  createdAt: string
  updatedAt: string
}

/** 需求创建请求 */
export interface RequirementCreateRequest {
  title: string
  description?: string
  priority?: Priority
  source?: string
  category?: string
  assignedTo?: number
  milestoneId?: number
}

/** 状态流转请求 */
export interface StatusUpdateRequest {
  status: RequirementStatus
  comment?: string
}
